python/src/spark_rapids_ml/core.py

```diff
@@ -400,6 +401,57 @@ def _require_nccl_ucx(self) -> Tuple[bool, bool]:
         """
         return (True, False)
 
+    def _validate_parameters(self) -> None:
+        cls_name = self.__class__.__name__
+
+        pyspark_est: Optional[JavaParams] = None
+
+        # Is there a better way for the dynamic imports?
+        if "LogisticRegression" in cls_name:
+            from pyspark.ml.classification import LogisticRegression
+
+            pyspark_est = LogisticRegression()
+        elif "RandomForestClassifier" in cls_name:
+            from pyspark.ml.classification import RandomForestClassifier
+
+            pyspark_est = RandomForestClassifier()
+        elif "RandomForestRegressor" in cls_name:
+            from pyspark.ml.regression import RandomForestRegressor
+
+            pyspark_est = RandomForestRegressor()
+        elif "LinearRegression" in cls_name:
+            from pyspark.ml.regression import LinearRegression
+
+            pyspark_est = LinearRegression()
+        elif "PCA" in cls_name:
+            from pyspark.ml.feature import PCA
+
+            pyspark_est = PCA()
+        elif "KMeans" in cls_name:
+            from pyspark.ml.clustering import KMeans
+
+            pyspark_est = KMeans()
+        if pyspark_est is not None:
+            # Both pyspark and cuml may have a parameter with the same name,
+            # but cuml might have additional optional values that can be set.
+            # If we transfer these cuml-specific values to the Spark JVM,
+            # it would result in an exception.
```

Comment:
Looks like we currently allow setting cuml param directly in which case we also set the corresponding Spark param, but to the cuml-specific value:  https://github.com/NVIDIA/spark-rapids-ml/blob/branch-23.08/python/src/spark_rapids_ml/params.py#L267
(see the TODO).   Not sure if this is actually used though.

Reply:
This is a good way to handle this issue.